### PR TITLE
chore(ci) make sure tags dont get built by the nightly cron

### PR DESCRIPTION
**Pull request by hutchic**
There's only two conditions in which Jenkins should be doing anything

Release:
- run from SCM don't run from cron

Nightly Release:
- run from timer don't run from SCM
- only run on master and next

SCM push to next / master is a noop
Capture

cron event runs build, test and nightly release stages
Capture2

SCM tag runs build, test and nightly release stage also the tag doesn't attempt to re-run every 24hrs
Capture3

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,17 +1,35 @@
 pipeline {
     agent none
+    triggers {
+        cron(env.BRANCH_NAME == 'master' | env.BRANCH_NAME == 'next' ? '@daily' : '')
+    }
```
Comment (hutchic) [inline Jenkinsfile]: we may need to cherry-pick and push this line onto the 2.x tags

File: Jenkinsfile [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
+                    allOf {
+                        buildingTag()
+                        not { triggeredBy 'TimerTrigger' }
+                    }
+                    allOf {
+                        triggeredBy 'TimerTrigger'
+                        anyOf { branch 'master'; branch 'next' }
+                    }
+                }
+            }
```
Comment (hutchic) [inline Jenkinsfile]: this evaluates to

```
if (
  ( buildingtag && not cron ) ||
  ( ( branch = master || branch = next) && cron )
)

Comment (kikito) [reply Jenkinsfile]: Please put that in a comment on the file, too.

Comment (hutchic) [reply Jenkinsfile]: added in f4485bbbaec09731108d6191dedaa6448702fdb6

**Pull request merged**